src/components/Select/Select.test.tsx

```diff
@@ -0,0 +1,564 @@
+  });
+
+  test("키보드로 옵션을 탐색할 수 있다", async () => {
+    const user = userEvent.setup();
+
+    render(
+      <Select placeholder="라이브러리를 선택하세요">
+        <option value="react">React</option>
+        <option value="vue">Vue</option>
+        <option value="svelte">Svelte</option>
+      </Select>,
+    );
+
+    const select = screen.getByRole("combobox");
+    select.focus();
+
+    await user.keyboard("{ArrowDown}");
+    await user.keyboard("{Enter}");
+
+    expect(screen.getByRole("option", { name: "React" })).toBeInTheDocument();
+  });
+
+  test("여러 옵션을 순차적으로 선택할 수 있다", async () => {
+    const user = userEvent.setup();
+
+    render(
+      <Select placeholder="라이브러리를 선택하세요">
+        <option value="react">React</option>
+        <option value="vue">Vue</option>
+        <option value="svelte">Svelte</option>
+      </Select>,
+    );
+
+    const select = screen.getByRole("combobox");
+
+    await user.selectOptions(select, "react");
+    expect(screen.getByRole("option", { name: "React" })).toBeInTheDocument();
+
+    await user.selectOptions(select, "vue");
+    expect(screen.getByRole("option", { name: "Vue" })).toBeInTheDocument();
+  });
+
+  test("clear 버튼을 클릭하면 값이 초기화된다", async () => {
+    const user = userEvent.setup();
+    const handleChange = vi.fn();
+
+    render(
+      <Select
+        value="react"
+        clearButtonName="clear"
+        onChange={handleChange}
+        placeholder="라이브러리를 선택하세요"
+      >
+        <option value="react">React</option>
+        <option value="vue">Vue</option>
+      </Select>,
+    );
+
+    const clearButton = screen.getByRole("button", {
+      name: "clear",
+    });
+    await user.click(clearButton);
+
+    expect(handleChange).toHaveBeenCalledWith(
+      expect.objectContaining({
+        target: expect.objectContaining({ value: "" }),
+      }),
+    );
+  });
+
+  test("clear 버튼 클릭 시 placeholder가 다시 표시된다", async () => {
+    const user = userEvent.setup();
+
+    const ControlledSelectTest = () => {
+      const [selectedValue, setSelectedValue] = useState<string>("react");
+
+      return (
+        <Select
+          value={selectedValue}
+          clearButtonName="clear"
+          onChange={(e) => {
+            setSelectedValue(e.target.value);
+          }}
+          placeholder="라이브러리를 선택하세요"
+        >
+          <option value="react">React</option>
+          <option value="vue">Vue</option>
+        </Select>
+      );
+    };
+
+    render(<ControlledSelectTest />);
+
+    expect(screen.getByRole("option", { name: "React" })).toBeInTheDocument();
+
+    const clearButton = screen.getByRole("button", {
+      name: "clear",
+    });
+    await user.click(clearButton);
+
+    await waitFor(() => {
+      const select: HTMLSelectElement = screen.getByRole("combobox");
+      expect(select.value).toBe("");
+    });
+    expect(screen.getByText("라이브러리를 선택하세요")).toBeInTheDocument();
+  });
+
+  test("disabled 상태일 때 clear 버튼이 작동하지 않는다", async () => {
```

Comment:
테스트 안에서 await을 사용하지 않아서 async 키워드는 필요 없겠네요!

```suggestion
  test("disabled 상태일 때 clear 버튼이 작동하지 않는다", () => {
```